### PR TITLE
Feat/niloufar mont

### PR DESCRIPTION
**Pull request by lilmont**
Pipeline tool:
> maha-builder --ip "someIp" --url "someUrl"

"localhost:xxxx/Products/GetUserById/[id]"
"localhost:xxxx/Products/GetAllProducts"

### CODE REVIEW

File: src/Niloo Mont/PipelineCLITool/Framework.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+namespace PipelineCLITool;
+public class Framework
+{
+    public void ExceptionHandling(HttpContext httpContext, Action<HttpContext> function)
+    {
+        try
+        {
+            if (httpContext.IP is "Iran")
+                throw new BannedIPException(httpContext.IP);
+            else
```
Comment (thisisnabi) [inline src/Niloo Mont/PipelineCLITool/Framework.cs:13]: You can remove this `else`

File: src/Niloo Mont/PipelineCLITool/Framework.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            {
+                userId = parts[3];
+
+                if (string.IsNullOrEmpty(actionName))
+                    throw new NoParameterProvidedException();
+
+                object[] parameters = new object[parameterList.Length];
+                for (int i = 0; i < parameterList.Length; i++)
+                {
+                    var convertToType = $"To{parameterList[i].ParameterType.Name}";
```
Comment (thisisnabi) [inline src/Niloo Mont/PipelineCLITool/Framework.cs:66]: Please move this logic in a helper class named model Binder for binding method parameters

File: src/Niloo Mont/PipelineCLITool/Program.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+﻿using Dumpify;
+using PipelineCLITool;
+
+Framework framework = new();
+
+string? ipArg = null;
+string? urlArg = null;
+
+for (int i = 0; i < args.Length; i++)
```
Comment (thisisnabi) [inline src/Niloo Mont/PipelineCLITool/Program.cs:9]: Move this code into the another method!
Don't write Spaghetti code

**Review by thisisnabi — CHANGES_REQUESTED** [review]

File: src/Niloo Mont/PipelineCLITool/Framework.cs (hunk 2 above)
Comment (lilmont) [reply src/Niloo Mont/PipelineCLITool/Framework.cs]: Done.

File: src/Niloo Mont/PipelineCLITool/Program.cs (hunk 3 above)
Comment (lilmont) [reply src/Niloo Mont/PipelineCLITool/Program.cs]: Sure, thank you for the great feedback.

**Review by thisisnabi — APPROVED** [review]

**Pull request merged**